pandera/typing/pandas.py

```diff
@@ -231,8 +232,15 @@ def from_records(  # type: ignore
         schema_index = schema.index.names if schema.index is not None else None
         if "index" not in kwargs:
             kwargs["index"] = schema_index
+        data_df = pd.DataFrame.from_records(data=data, **kwargs)
         return DataFrame[schema](  # type: ignore
-            pd.DataFrame.from_records(data=data, **kwargs,)[
-                schema.columns.keys()
-            ]  # set the column order according to schema
+            # set the column order according to schema
+            data_df[
+                list(
+                    filter(
+                        lambda column: column in data_df.columns,
+                        OrderedDict.fromkeys(schema.columns),
```

Comment:
isn't insertion order preserved as of python3+? the filter here is fine, but I don't think we need the extra import here.

Reply:
You are totally right. From python 3.7+ order is preserved and pandera is set for python 3.7 onwards.